### ISSUE TITLE
journalctl --setup-keys: If filesystem does not support NOCOW, other attributes are also not set

### ISSUE THREAD

**Issue by vuzdemav** [systemd/systemd#16203]
**systemd version the issue has been seen with**

> systemd 245 (245.4-4ubuntu3.1)
> +PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid

**Used distribution**
 Ubuntu 20.04 virtual machine in KVM, based on Ubuntu cloud image
https://cloud-images.ubuntu.com/releases/focal/release-20200513/ubuntu-20.04-server-cloudimg-amd64.img

The same problem is present in Ubuntu 18.04

**Expected behaviour you didn't see**
Setting NOCOW fails but the other requested attributes are set:
```
journalctl --setup-keys
[ ... ]
Failed to set file attributes: Operation not supported
[ ... ]
lsattr /var/log/journal/*/fss 
s-S---d-------e----- /var/log/journal/5a04c279cf034fb08c01091104ba5200/fss
```

**Unexpected behaviour you saw**
No attributes have been set:
```
journalctl --setup-keys
[ ... ]
Failed to set file attributes: Operation not supported
[ ... ]
lsattr /var/log/journal/*/fss
--------------e----- /var/log/journal/5a04c279cf034fb08c01091104ba5200/fss
```

**Steps to reproduce the problem**
Run `journalctl --setup-keys`, with /var/log on a filesystem which does not support the NOCOW attribute (e.g. ext4)

**Comment by vuzdemav** [comment]
This is the `chattr_fd` call: https://github.com/systemd/systemd/blob/120b5c0bbead6f635eeebce0a47ec5a777560f74/src/journal/journalctl.c#L1871

**Comment by poettering** [comment]
That's fixed coincidentally by #16145
